File: app/src/main/java/com/kenny/separatededittextdemo/MainActivity.java
package com.kenny.separatededittextdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.kenny.separatededittext.SeparatedEditText;

public class MainActivity extends AppCompatActivity {

    SeparatedEditText solid;
    SeparatedEditText hollow;
    SeparatedEditText underline;

    boolean showContent;
    boolean showCursor = true;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        solid = (SeparatedEditText) findViewById(R.id.edit_solid);
        hollow = (SeparatedEditText) findViewById(R.id.edit_hollow);
        underline = (SeparatedEditText) findViewById(R.id.edit_underline);

    }


    public void handleContent(View v) {
        solid.setPassword(!showContent);
        hollow.setPassword(!showContent);
        underline.setPassword(!showContent);
        showContent = !showContent;
    }

    public void handleCursor(View v) {
        solid.setShowCursor(!showCursor);
        hollow.setShowCursor(!showCursor);
        underline.setShowCursor(!showCursor);
        showCursor = !showCursor;
    }

}

File: app/src/main/java/com/kenny/separatededittextdemo/MainActivity.kt
package com.kenny.separatededittextdemo

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.kenny.separatededittext.SeparatedEditText
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var showContent = false
    var showCursor = true
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        edit_solid.setTextChangedListener(object : SeparatedEditText.TextChangedListener {
            override fun textChanged(changeText: CharSequence?) {
            }

            override fun textCompleted(text: CharSequence?) {
                edit_solid.showError()
            }

        })
        edit_underline.setTextChangedListener(object : SeparatedEditText.TextChangedListener {
            override fun textChanged(changeText: CharSequence?) {
            }

            override fun textCompleted(text: CharSequence?) {
                edit_underline.showError()
            }

        })
    }

    fun handleContent(v: View?) {
        edit_solid.setPassword(!showContent)
        edit_hollow.setPassword(!showContent)
        edit_underline.setPassword(!showContent)
        showContent = !showContent
    }

    fun handleCursor(v: View?) {
        edit_solid.setShowCursor(!showCursor)
        edit_hollow.setShowCursor(!showCursor)
        edit_underline.setShowCursor(!showCursor)
        showCursor = !showCursor
    }
}